File: ILT.IHR.DTO/Expense.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ILT.IHR.DTO
{
    public partial class Expense : AbstractDataObject
    {
        public int ExpenseID
        {
            get { return base.RecordID; }
            set { base.RecordID = value; }
        }
        [Required]
        public int? EmployeeID { get; set; }
        public string EmployeeName { get; set; }
        [Required]
        public int ExpenseTypeID { get; set; }
        public string ExpenseType { get; set; }
        public string FileName { get; set; }
        [Required]
        public decimal? Amount { get; set; }
        [Required]
        public DateTime SubmissionDate { get; set; }
        public string SubmissionComment { get; set; }
        public int StatusID { get; set; }

        public Guid LinkID { get; set; }
        public string Status { get; set; }
        public int? AmountPaid { get; set; }
        public DateTime? PaymentDate { get; set; }
        public string PaymentComment { get; set; }
    }
}

File: ILT.IHR.APP/ClientApp/src/app/core/interfaces/Expense.ts
export class IExpense {
    ExpenseID: number;
    EmployeeID: number;
    EmployeeName: String;
    ExpenseTypeID: number;
    ExpenseType: string;
    FileName: string;
    Amount?: number;
    SubmissionDate: Date;
    SubmissionComment: string;
    StatusID: number;
    LinkID: string;
    Status: string;
    AmountPaid?: number;
    PaymentDate?: Date;
    PaymentComment: string;
    CreatedBy: string;
    CreatedDate: Date;
    ModifiedBy: string;
    ModifiedDate: Date;
    TimeStamp: ['']
}

export class IExpenseForDisplay {
    expenseID: number;
    employeeID: number;
    employeeName: String;
    expenseTypeID: number;
    expenseType: string;
    fileName: string;
    amount?: number;
    submissionDate: Date;
    submissionComment: string;
    statusID: number;
    linkID: string;
    status: string;
    amountPaid?: number;
    paymentDate?: Date;
    paymentComment: string;
    createdBy: string;
    createdDate: Date;
    modifiedBy: string;
    modifiedDate: Date;
    timeStamp: ['']
}
